### ISSUE TITLE
The Last Milestone

### ISSUE THREAD

**Issue by bilal-24** [githubharald/WordDetectorNN#20]
Hello,
I have a problem with my model and I don't know how to solve it. I want to build a model which doesn't take commas or other small punctuation marks. Could you please help me to figure out how to do that?

Here's the situation:

Some parameters and hyperparameters were changed, for example:

The thresh value in eval.py,
DBSCAN(min_samples, eps), (xmin, xmax, ymin, ymax) values in aabb_clustering.py,
max_aabbs in train.py(during training with my data), infer.py, eval.py

There are also some added functions:

1) To sort the words in the same row. 
2) To prevent the nested words on the x-axis.

The model has been trained with ResNet34, but there are no significant differences between ResNet18 & ResNet34 results.

I'd like to thank you because you helped me a lot with this project, your answers were very helpful.
I hope your answer will help to solve my last little issue. 
And of course, despite you even don't need it, I see it as a debt to mention you in my project.

Best regards

**Comment by githubharald** [comment]
Hi,
what exactly do you mean with "doesn't take commas or other small punctuation marks"? Do you want that the model does not detect a separate bounding box around a punctuation mark, or do you want that the bounding box around a word should not be too large so that close punctuation marks are not included?

Taken from your image, what do you want to have in this case, how should the detection be changed:
![grafik](https://user-images.githubusercontent.com/15148095/185619805-85ad0b0a-3b63-444a-bca6-ccc02f6f3d78.png)

By the way, in this repo https://github.com/githubharald/WordDetector there is a function `sort_line` which clusters the words into lines and sorts the words in each line, that might also be useful for you.

**Comment by bilal-24** [comment]
Hello,
Thank you for your answer. I don't want to take the punctuation marks at all, in any bounding box.

I don't want these situation:

![comma_1](https://user-images.githubusercontent.com/76588928/185849678-6cd00c1e-0126-4f28-b4f5-6fa8d9d652e2.png)
![comma_2](https://user-images.githubusercontent.com/76588928/185849703-b0fce7a2-0c92-40f1-99ad-24f71e18384a.png)
![dot_1](https://user-images.githubusercontent.com/76588928/185849714-84f1961e-1096-415b-908a-6e1160259761.png)
![dot_comma_1](https://user-images.githubusercontent.com/76588928/185849721-51f5bc0e-f7ad-4f93-9f3d-7c4c42ae4e92.png)

I want to take these results:

![comma_want_1](https://user-images.githubusercontent.com/76588928/185849823-d3c548b0-e2b3-4eb8-b0f3-598b5bb078c6.png)
![comma_want_2](https://user-images.githubusercontent.com/76588928/185849830-abb60e22-8938-46ec-b843-e6437a53deca.png)
![dot_want_1](https://user-images.githubusercontent.com/76588928/185849847-322db833-53c9-48e3-a4e9-19e8e62f5906.png)

**Comment by githubharald** [comment]
I don't have the training data (IAM) at hand currently, please check if the punctuation marks have separate bounding boxes. If yes, the model should learn it, otherwise, the training data would have to be adapted.

If the training data is fine, next I would have a look into the bounding box clustering. Maybe the model correctly predicts the bounding box around the punctuation marks and also around the words, but the clustering algo then merges them. If so, the clustering has to be adapted, e.g. by also considering the bounding box size so that small and large bounding boxes don't get merged.
